File: 프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.py
from collections import Counter
def solution(weights):
    answer = 0
    weight_count = Counter(weights)
    for w in weight_count:
        answer += (weight_count[w] * (weight_count[w]-1)) / 2
        answer += weight_count[w] * weight_count[w*(4/3)]
        answer += weight_count[w] * weight_count[w*(4/2)]
        answer += weight_count[w] * weight_count[w*(3/2)]
    return answer

File: 프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.js
function solution(weights) {
    let answer = 0;
    const weightCount = new Map()
    for (const w of weights) {
        weightCount.set(w, (weightCount.get(w) || 0) + 1);
    }
    
    for (const [w, count] of weightCount.entries()) {
        answer += (count * (count-1)) / 2;
        const ratios = [4/3,2/3,2];
        for (const r of ratios) {
            const target = w * r;
            if (weightCount.has(target)) {
                answer += count * weightCount.get(target)
            }
        }
    }
    
    return answer;
}